src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -314,6 +316,56 @@ public Requester contentType(String contentType) {
         return this;
     }
 
+    /**
+     * NOT FOR PUBLIC USE. Do not make this method public.
+     * <p>
+     * Sets the path component of api URL without URI encoding.
+     * <p>
+     * Should only be used when passing a literal URL field from a GHObject, such as {@link GHContent#refresh()} or when
+     * needing to set query parameters on requests methods that don't usually have them, such as
+     * {@link GHRelease#uploadAsset(String, InputStream, String)}.
+     *
+     * @param urlOrPath
+     *            the content type
+     * @return the requester
+     */
+    Requester setRawUrlPath(String urlOrPath) {
```

Comment:
Eventually, we'll want to call this method directly where it is needed, but for now `withUrlPath` has logic to decide when to use this.